src/python/WMCore/WMSpec/Steps/Executors/DQMUpload.py

```diff
@@ -150,27 +150,33 @@ def upd(m, data):
         logging.info(msg)
 
         try:
+            logException = True
             for uploadURL in self.step.upload.URL.split(';'):
                 (headers, data) = self.upload(uploadURL, args, filename)
-                msg = 'HTTP upload finished succesfully with response:\n'
-                msg += 'Status code: %s\n' % headers.get("Dqm-Status-Code", None)
-                msg += 'Message: %s\n' % headers.get("Dqm-Status-Message", None)
-                msg += 'Detail: %s\n' % headers.get("Dqm-Status-Detail", None)
-                msg += 'Data: %s\n' % str(data)
-                logging.info(msg)
+                msg = '  Status code: %s\n' % headers.get("Dqm-Status-Code", None)
+                msg += '  Message: %s\n' % headers.get("Dqm-Status-Message", None)
+                msg += '  Detail: %s\n' % headers.get("Dqm-Status-Detail", None)
+                msg += '  Data: %s\n' % str(data)
+                if int(headers.get("Dqm-Status-Code", 400)) >= 400:
```

Comment:
I assume it doesn't contain the Dqm-Status-Code considers as failure?

Reply:
Yes, that's why I put the default to 400. If it doesn't have that header, then - I believe - the DQM server didn't handle that request.